lib/rust/lexer/definition/src/token.rs

```diff
@@ -336,9 +621,50 @@ impl Shape {
         Shape::DanglingBase(base.into())
     }
 
-    /// Construct a text literal.
-    pub fn text(text:impl Into<String>) -> Shape {
-        Shape::Text(text.into())
+    /// Construct a text line literal.
+    pub fn text_line(style:TextStyle, segments:Vec<Token>) -> Shape {
+        Shape::TextLine{style,segments}
+    }
+
+    /// Construct an inline block text literal.
+    pub fn text_inline_block(style:TextStyle, segments:Vec<Token>) -> Shape {
+        Shape::TextInlineBlock{style,segments}
+    }
+
+    /// Construct a text block literal.
+    pub fn text_block(start_line_ending: LineEnding, style:TextStyle, lines:Vec<Token>) -> Shape {
+        Shape::TextBlock{start_line_ending,style,lines}
+    }
+
+    /// Construct an invalid quote literal.
+    pub fn invalid_quote(bad_quotes:impl Str) -> Shape {
+        Shape::InvalidQuote(bad_quotes.into())
+    }
+
+    /// Construct a raw text segment.
+    pub fn text_segment_raw(text:impl Str) -> Shape {
+        Shape::TextSegmentRaw(text.into())
+    }
+
+    /// Construct a text segment containing an escape sequence.
+    pub fn text_segment_escape(style:EscapeStyle, repr_str:impl Str) -> Shape {
+        let repr = repr_str.into();
+        Shape::TextSegmentEscape{style,repr}
+    }
```

Comment:
sometimes you are doing smart constructors as camel case functions, sometimes this way. Lets justu se one way everywhere

Reply:
The difference is that here they would clash with the actual definitions in the enum, but in `Token` there is no such clash to occur and hence it makes more sense to pretend that they're constructors.

Reply:
I understand the reason, but more important is consistency - lets use everywhere one style.